src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs

```diff
@@ -210,10 +210,26 @@ private string GetConnectionString(ConnectionInfo connInfo, string databaseName)
             return ConnectionService.BuildConnectionString(connInfo.ConnectionDetails);
         }
 
-        private static string RemoveExcessWhitespace(string script)
+        public static string RemoveExcessWhitespace(string script)
         {
-            // replace all multiple spaces with single space
-            return Regex.Replace(script, " {2,}", " ");
+            if(script != null)
```

Comment:
as a style note, we should add a ` ` between `f` and `(` in `if (` statements to be consistent with this codebase.